addons/account/models/account_journal_dashboard.py

```diff
@@ -349,12 +349,35 @@ def _fill_bank_cash_dashboard_data(self, dashboard_data):
                    ) statement ON TRUE
              WHERE journal.id = ANY(%s)
                AND statement.company_id = ANY(%s)
-        """, [self.ids, self.env.companies.ids])
+        """, [bank_cash_journals.ids, self.env.companies.ids])
         last_statements = {journal_id: statement_id for journal_id, statement_id in self.env.cr.fetchall()}
-        self.env['account.bank.statement'].browse(i for i in last_statements.values() if i).mapped('balance_end_real')  # prefetch
+        self.env['account.bank.statement'].browse(i for i in last_statements.values() if i).mapped(lambda s: (s.balance_end_real, s.date))  # prefetch
 
         outstanding_pay_account_balances = bank_cash_journals._get_journal_dashboard_outstanding_payments()
 
+        # Misc Entries (journal items in the default_account not linked to bank.statement.line)
+        misc_domain = []
+        for journal in bank_cash_journals:
+            statement = self.env['account.bank.statement'].browse(last_statements.get(journal.id))
+            date_limit = statement.date or journal.company_id.fiscalyear_lock_date
+            misc_domain.append(
+                date_limit
+                and [('account_id', '=', journal.default_account_id.id), ('date', '>', date_limit)]
+                or [('account_id', '=', journal.default_account_id.id)]
```

Comment:
What about this @h4818 

```suggestion
                [('account_id', '=', journal.default_account_id.id), ('date', '>', date_limit)]
                if date_limit else
                [('account_id', '=', journal.default_account_id.id)]
```